### DISCUSSION TITLE
Failed with yarprobotinterface routine after robotology-superbuild update

### DISCUSSION THREAD

**Discussion by justdiandes** [robotology/community#707] [Q&A]
We tried to update the iCub from commit `683ef9669c7224046b28044005edecb61e476e52` to the `v2024.02.0` in both a backup laptop and `icub-head`. In that configuration it's possible to establish connections in both `icub-head` and `icubsrv`, but when the `yarprobotinterface` module is initialized from `yarpmanager`, the iCub does not respond. For more information related to the robot, our iCub is **iCubValparaiso01**, and we never made some kind of update related to the robotology-superbuild packages.

The `yarplogger` shows the following errors:

![Screenshot from 2024-06-05 16-20-43](https://github.com/robotology/community/assets/121194422/5a96018f-8a42-4214-b4fa-a7acf324a393)

And the same messages for every part of the body, so the question here is if this issue is related to some firmware update that we need to do first, or if some problem related to the laptop configuration. We also noted that the wrappers files for the FT sensors changed from the oldest version that we had, so we were wondering if these configurations are setting up automatically or if we need to edit some files.

We also wanted to know if there is a way to make sure that the robot is correctly updated. Our aim is to make sure that if the communication between the laptop and icub persists, could be related to the laptop rather than the iCub per se.

**Comment by Nicogene** [comment]
Hi @justdiandes,

could you please post the txt file of the log?
From the screenshot is difficult to understand the error, my hunch is that after upgrading the software in `icub-head` you have triggered an incompatibility with/ the firmware.

If you want to use that distro you need to upgrade the firmware accordingly. You can do it via `FirmwareUpdater`:
- https://icub-tech-iit.github.io/documentation/icub_firmware/firmwareupdater/firmwareupdater/

Or you can try the automatic procedure that has been recently revamped (cc @valegagge @davidetome)

**Reply by justdiandes** [reply]
Sure! 
[log_icub-head_yarprobotinterface_06-05.txt](https://github.com/user-attachments/files/15642367/log_icub-head_yarprobotinterface_06-05.txt)

There it is the full log output. I'm gonna check on the solutions related to the firmware update, but I'll keep an eye on any feedback you can provide

**Reply by justdiandes** [reply]
Hi @Nicogene!

As an update of the situation, we updated the firmware and the robot responded correctly to the yarprobotinterface routine and also we could initialized the other modules, such as the `wholeBodyDynamics`, that cause some errors earlier. But now we are facing kind of the same issue as the #[621](https://github.com/orgs/robotology/discussions/621), with the cameras. We tried to update the XML of the App and the cameras only show the first frame that they can take and then it freeze. If you have any recommendations that would be helpfull. Otherwise, we are going to continue with the updating process of the module trying to fix this issue

**Reply by justdiandes** [reply]
Well, from the previous issue we managed to fix the problem of the cameras by updating the `cameras_calib.xml` in the yarpdev parts. Also we needed to change the connection that was initially in `udp` protocol, to `tcp`. This fix the issue and the image from the cameras is working correctly. 

From that thing, we had problemas in all Apps, in the following one, the emotions, we fix that also updating the XML by changing the `yarpdev` module to a `yarprobotinterface` one, as the suggestion in [this issue](https://github.com/robotology/icub-main/pull/883/commits/1df4e8fd1ff39633225e5785772b79b6f507c0e5). It works but makes me wonder if there's any way to update all this Apps to the corresponding version, because now we have some issues with the skinManager, where the parameters of the module are the following:

```
<parameters>--from skinManAllPlusLegs.ini --context skinGui</parameters>
```
so now we cannot make the connection between the skin sensors and the skinGui.

Also something else that we saw is that the ports `wholeBodyDynamics/right_arm/FT:i` and `wholeBodyDynamics/left_armFT:i` are not initialized, so the `Red-Ball_Demo` is working with problems and we don't know if it is related in som kind of way.

Like we mentioned before, we updated the `robotology-superbuild` to the version `v2024.02.0` and we don't know if this issue is related to this or some other problem.

**Reply by Nicogene** [reply]
> It works but makes me wonder if there's any way to update all this Apps to the corresponding version, because now we have some issues with the skinManager,

Unfortunately, there is not, but for the `skinManager` you do not need `yarprobotinterface`, it can be launched as a standalone executable.

You are experiencing issues on a lot of executables it is complicated to understand the source of the error without the logs, could you please post the log of all the failing processes?

**Reply by justdiandes** [reply]
Hi,
I understand, I'm gonna get the log output from this apps, but we are facing some issues related to the calibration steps in [here](https://icub-tech-iit.github.io/documentation/icub_robot_calibration/icub-robot-calibration-v2.x/). This issue is new to us and could be related to the previews updating that we did, 'cause we didn't have the problem in the old robotology-superbuild version.

The output in this steps are the following:

[out.txt](https://github.com/user-attachments/files/15993405/out.txt)

I hope that this txt is enough to scan the issue.

For the problems related to the Apps, I did some edits on the xml files and some Apps are now working just fine.

**Reply by Nicogene** [reply]
> but we are facing some issues related to the calibration steps in [here](https://icub-tech-iit.github.io/documentation/icub_robot_calibration/icub-robot-calibration-v2.x/)

Why are you re-doing the robot calibration? Usually, this is a procedure that is done in the production, that has to be done in case of problems in the calibration.

In particular, what issue are you facing? The problem is that `yarprobotinterface` is not starting?
The curious thing that I see from your log is that there are no errors.

cc @valegagge @marcoaccame
